corehq/apps/domain/templates/domain/confirm_billing_info.html

```diff
@@ -7,31 +7,14 @@
     <script src="{% static 'accounting/js/widgets.js' %}"></script>
     <script src="https://js.stripe.com/v2/"></script>
     <script src="{% static 'accounting/js/stripe_card_manager.js' %}"></script>
-{% endblock %}
-
-
-{% block js-inline %}{{ block.super }}
-    <script>
-        $('a.breadcrumb-2').click(function (e) {
-            e.preventDefault();
-            var url = $(this).attr('href');
-            var $navigateForm = $('<form method="post" style="display: none;" />').attr('action', url + 'confirm/');
-            $(this).after($navigateForm);
-            $navigateForm.append($('<input type="hidden" name="plan_edition" />').val('{{ plan.edition }}'));
-            $navigateForm.submit();
-        });
-    </script>
-    <script>
-        Stripe.setPublishableKey('{{ stripe_public_key }}');
-        var cardManager = new hqImport("accounting/js/stripe_card_manager").StripeCardManager({
-            cards: {{ cards|JSON }},
-            url: "{% url 'cards_view' domain %}"
-        });
-        $("#card-manager").koApplyBindings(cardManager);
-    </script>
+    <script src="{% static 'domain/js/confirm_billing_info.js' %}"></script>
 {% endblock %}
 
 {% block form_content %}
+    {% initial_page_data "plan" plan %}
+    {% initial_page_data "cards" cards %}
+    {% initial_page_data "stripe_public_key" stripe_public_key %}
+    {% initial_page_data "cards_view_url" url 'cards_view' domain %}
```

Comment:
For URLs in particular, there's a `registerurl` tag instead of `initial_page_data` - same general idea, but `reigsterurl` allows js to pass in url arguments. https://github.com/dimagi/js-guide/blob/master/integration-patterns.md#django-urls

Reply:
Nice